argocd/resource_argocd_application.go

```diff
@@ -92,11 +93,15 @@ func resourceArgoCDApplicationCreate(d *schema.ResourceData, meta interface{}) e
 	}
 	d.SetId(app.Name)
 	if wait, ok := d.GetOk("wait"); ok && wait.(bool) {
-		return resource.Retry(d.Timeout(schema.TimeoutCreate), func() *resource.RetryError {
+		err = resource.RetryContext(context.Background(), d.Timeout(schema.TimeoutCreate), func() *resource.RetryError {
 			a, err := c.Get(context.Background(), &applicationClient.ApplicationQuery{
 				Name: &app.Name,
 			})
 			if err != nil {
+				fmt.Printf("%v\n", err)
```

Comment:
You should pass the error on to Terraform error handling, or the new diag.Diagnostics introduced in the sdk v2. Not print an error on the standard output.